src/main/java/com/gmail/nossr50/listeners/PlayerListener.java

```diff
@@ -789,12 +789,12 @@ public void onPlayerInteractMonitor(PlayerInteractEvent event) {
         if(event.getAction() == Action.RIGHT_CLICK_BLOCK || event.getAction() == Action.RIGHT_CLICK_AIR)
         {
             if(ExperienceConfig.getInstance().isFishingExploitingPrevented()
-                       && (heldItem.getType() == Material.FISHING_ROD || player.getInventory().getItemInOffHand().getType() == Material.FISHING_ROD))
+                       && (heldItem.getType() == Material.FISHING_ROD || player.getInventory().getItemInOffHand().getType() == Material.FISHING_ROD)
+                       && !mcMMO.getCompatibilityManager().hasPatchedMinecartFlight())
             {
                 if(player.isInsideVehicle() && (player.getVehicle() instanceof Minecart || player.getVehicle() instanceof PoweredMinecart))
```

Comment:
`PoweredMinecart` is already an extension of `Minecart`. So this check can be simplified.

Also, this may just be me but I feel like the Minecart Flight patch check should go into the if-statement with the Minecart check, not the one with the fishing rod.
It just seems like it makes more sense to me that way.

```suggestion
                       && (heldItem.getType() == Material.FISHING_ROD || player.getInventory().getItemInOffHand().getType() == Material.FISHING_ROD))
            {
                if (player.isInsideVehicle() && player.getVehicle() instanceof Minecart
                    && !mcMMO.getCompatibilityManager().hasPatchedMinecartFlight())
```